Refactoring of jobs and deployment parts

gfieni opened the pull request:
PR overview :
- Cleaning functions and adding comments for 'Job' and 'Deployment' parts ;
- Moving flags value checks in the flags processing function ;
- Moving JSON requests from string to structures for marshal/unmarshal ;
- Moving the job reservation and node deployment parts in the driver to be able to reuse it easily ;
- Adding functions to wait until the job/deployment is finished ;
- Adding a function for getting a deployment informations from its id ;
- Removing the function for reading the ssh public key => use of golang 'ioutil.ReadFile' ;
- Updating README.

gfieni committed:
refactor(driver): Moving flags value checks in the flags processing
function.

gfieni committed:
refactor(jobs): Cleaning job submission part

Moving jobs submission logging parts inside SubmitJob function.
Moving job request creation outside of SubmitJob function.
Moving unrelated to API function to a 'utils' file.
Adding comments to job submission parts.

gfieni committed:
refactor(jobs): Cleaning GetJob function

gfieni committed:
refactor(jobs): Cleaning GetJobState function

The job state (running, terminated) is already handled by the API, no need to calculate the end of the job.
Removing the function jobIsOver.

gfieni committed:
refactor(jobs): Cleaning KillJob function

gfieni committed:
refactor(jobs): Cleaning and renaming WaitJobIsReady function

BREAKING CHANGE: renaming function waitJobIsReady to WaitUntilJobIsReady

gfieni committed:
refactor(deployment): Cleaning and refactoring of the deployment parts

Renaming function DeployEnvironment to SubmitDeployment
Moving waiting for a deployment to finish part in a function WaitUntilDeploymentIsFinished
Moving deployment creation outside of the submission part
Moving deployment update status in a function GetDeployment (related to WaitUntilDeploymentIsFinished)
Moving logging parts to their related functions

BREAKING CHANGE: Rename of DeployEnvironment to SubmitDeployment

gfieni committed:
docs(README): Adding modifications asked in PR #6

mcolmant commented:
During test, `docker-machine` may blocks sometimes after the provisioning. After investigations, it does not seem to be related to the driver itself, but to the `docker-machine` command.

gfieni committed:
refactor(driver): Add log message when deployment finish.

gfieni committed:
refactor(deployment): Add some logs message when waiting for a job/deployment to run/finish

Moving logs from driver to WaitUntilDeploymentIsFinished function
Adding logs to WaitUntilJobIsReady function
